test/fakeintake/server/server.go

```diff
@@ -390,3 +395,45 @@ func (fi *Server) handleGetRouteStats(w http.ResponseWriter, req *http.Request)
 		body:        jsonResp,
 	})
 }
+
+// handleConfigureOverride sets a hardcoded HTTP response for requests to a particular endpoint
+func (fi *Server) handleConfigureOverride(w http.ResponseWriter, req *http.Request) {
+	if req == nil {
+		response := buildErrorResponse(errors.New("invalid request, nil request"))
+		writeHTTPResponse(w, response)
+		return
+	}
+
+	if req.Method != http.MethodPost {
+		response := buildErrorResponse(fmt.Errorf("invalid request method %s", req.Method))
+		writeHTTPResponse(w, response)
+		return
+	}
+
+	if req.Body == nil {
+		response := buildErrorResponse(errors.New("invalid request, nil body"))
+		writeHTTPResponse(w, response)
+		return
+	}
+
+	var payload api.ResponseOverride
+	err := json.NewDecoder(req.Body).Decode(&payload)
+	if err != nil {
+		log.Printf("Error reading body: %v", err.Error())
+		response := buildErrorResponse(err)
+		writeHTTPResponse(w, response)
+		return
+	}
+
+	log.Printf("Handling configureOverride request for endpoint %s", payload.Endpoint)
+
+	fi.responseOverrides[payload.Endpoint] = httpResponse{
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
You'll need a mutex to protect concurrent access in read / write mode

Reply:
Done!